src/protocol/context.rs

```diff
@@ -63,6 +65,31 @@ impl<'a, F: Field, SS: SecretSharing<F>> ProtocolContext<'a, SS, F> {
             prss: self.prss,
             gateway: self.gateway,
             accumulator: self.accumulator.clone(),
+            record_id: self.record_id,
+            _marker: PhantomData::default(),
+        }
+    }
+
+    #[must_use]
+    /// Make a sub-context which is bound to a record in case the same step is bound to a different `record_id`
+    /// # Panics
+    /// Panics in case the context is already bound to the same `record_id`
+    pub fn bind(&self, record_id: RecordId) -> Self {
```

Comment:
Thanks @akoshelev for helping with a sketch to introduce a bind to bind a step to a record in case it is already bound to a different record. @martinthomson would this work for now while we find a better solution?

Reply:
@martinthomson is travelling to London this week. This looks like an acceptable compromise to me (for now), although as we have discussed it does not provide us with sufficient protection against re-use of the same PRSS context the generate shared randomness multiple times. We can address that in another PR, e.g. #208